internal/kgateway/proxy_syncer/metrics.go

```diff
@@ -28,33 +29,95 @@ var (
 			Subsystem:                       statusSubsystem,
 			Name:                            "status_sync_duration_seconds",
 			Help:                            "Status sync duration",
+			Buckets:                         statusSyncHistogramBuckets,
 			NativeHistogramBucketFactor:     1.1,
 			NativeHistogramMaxBucketNumber:  100,
 			NativeHistogramMinResetDuration: time.Hour,
 		},
 		[]string{syncerNameLabel},
 	)
-	statusSyncResources = metrics.NewGauge(
+
+	transformsHistogramBuckets = []float64{0.0005, 0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5}
+	snapshotTransformsTotal    = metrics.NewCounter(
+		metrics.CounterOpts{
+			Subsystem: snapshotSubsystem,
+			Name:      "transforms_total",
+			Help:      "Total number of XDS snapshot transforms",
+		},
+		[]string{"gateway", "namespace", "result"},
+	)
+	snapshotTransformDuration = metrics.NewHistogram(
+		metrics.HistogramOpts{
+			Subsystem:                       snapshotSubsystem,
+			Name:                            "transform_duration_seconds",
+			Help:                            "XDS snapshot transform duration",
+			Buckets:                         transformsHistogramBuckets,
+			NativeHistogramBucketFactor:     1.1,
+			NativeHistogramMaxBucketNumber:  100,
+			NativeHistogramMinResetDuration: time.Hour,
+		},
+		[]string{"gateway", "namespace"},
+	)
+	snapshotResources = metrics.NewGauge(
 		metrics.GaugeOpts{
-			Subsystem: statusSubsystem,
+			Subsystem: snapshotSubsystem,
 			Name:      "resources",
-			Help:      "Current number of resources managed by the status syncer",
+			Help:      "Current number of resources in XDS snapshot",
 		},
-		[]string{syncerNameLabel, "name", "namespace", "resource"},
+		[]string{"gateway", "namespace", "resource"},
+	)
+
+	resourcesHistogramBuckets          = []float64{0.1, 0.5, 1, 2.5, 5, 10, 30, 60, 120, 300, 600, 1200, 1800}
+	resourcesStatusSyncsCompletedTotal = metrics.NewCounter(
+		metrics.CounterOpts{
+			Subsystem: resourcesSubsystem,
+			Name:      "status_syncs_completed_total",
+			Help:      "Total number of status syncs completed for resources",
+		},
+		[]string{"gateway", "namespace", "resource"})
+	resourcesStatusSyncDuration = metrics.NewHistogram(
+		metrics.HistogramOpts{
+			Subsystem:                       resourcesSubsystem,
+			Name:                            "status_sync_duration_seconds",
+			Help:                            "Initial resource update until status sync duration",
```

Comment:
Does "initial" mean only the first time? is the duration for the sync? or maybe something like:

```suggestion
			Help:                            "Duration of time for a resource update to be synced in the status",
```

Reply:
And relatedly, this one is the duration between the initial resource update and it receiving a status report. This is related to when the status is available on the resource in the cluster. I changed the Help text to:  "Duration of time for a resource update to receive a status report" Do you think that works?

Reply:
yes, thanks for making these changes!